### PR TITLE
GAL-G5-AUX

### PR DESCRIPTION
**Pull request by dari0rm**
## This branch has all of the necessary changes to make the application functional again after the mayor DB structure overhaul, there are also some minor changes that have been picked out during the process.

IMPORTANT: Conflicts are being resolved!

- [x] Conflicts resolved

### Changes
- [x] Reduced pack new name validator length to 40 to improve pack detail styles
- [x] Removed date begin and end from pack home card
- [x] Removed date begin and end from pack detail 
### Fixes
- [x] Fixed broken pagination query
- [x] Fixed databinding to match new DB structure
- [x] Fixed broken redirect when accessing a pack via URL, this redirect would send the user to my packs causing an error if the user's role did not have the required permissions 
- [x] Fixed packDetailquery, which was returing duplicated data when a pack had more than one entry on the pack_date table

### CODE REVIEW

File: cd2024bfs2g1-frontend/src/main/ngx/src/assets/i18n/en.json [hunk 1]
```diff
@@ -321,5 +321,7 @@
   "A POBRA DO CARAMINAL": "A POBRA DO CARAMIÑAL",
   "AS PONTES DE GARCIA RODRIGUEZ": "AS PONTES DE GARCÍA RODRÍGUEZ",
   "SAN SADURNINO": "SAN SADURNIÑO",
-  "VALDOVINO": "VALVODIÑO"
+  "VALDOVINO": "VALVODIÑO",
```
Comment (cesarsantiagonunez) [inline cd2024bfs2g1-frontend/src/main/ngx/src/assets/i18n/en.json:324]: faltan traduccións de idiomas

File: cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java [hunk 2]
```diff
@@ -86,13 +88,31 @@ public EntityResult packClientQuery(Map<String, Object> keysValues, List<String>
     public EntityResult allPacksQuery(Map<String, Object> keysValues, List<String> attributes) throws OntimizeJEERuntimeException {
         return this.daoHelper.query(this.packDao, keysValues, attributes, this.packDao.PCK_ALL_QUERY);
     }
+    @Override
+    public EntityResult packDetailQuery(Map<String, Object> keysValues, List<String> attributes) throws OntimizeJEERuntimeException {
+        if(keysValues.size()>0){
```
Comment (cesarsantiagonunez) [inline cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java:93]:
```suggestion
        if(keysValues.containsKey(PackDao.PCK_ID)){
```

File: cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java [hunk 3]
```diff
@@ -86,13 +88,31 @@ public EntityResult packClientQuery(Map<String, Object> keysValues, List<String>
...
+            Object key = keysValues.remove("pck_id");
```
Comment (cesarsantiagonunez) [inline cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java:94]: utilizade os placeholders

File: cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java [hunk 4]
```diff
@@ -86,13 +88,31 @@ public EntityResult packClientQuery(Map<String, Object> keysValues, List<String>
...
     @Override
     @Secured(PermissionsProviderSecured.SECURED)
     @Transactional(rollbackFor = Throwable.class)
     public EntityResult packInsert(Map<String, Object> attrMap) throws OntimizeJEERuntimeException, ParseException {
         Object imgCode = attrMap.get(ImageDao.ATTR_IMAGE_CODE);
         attrMap.remove(ImageDao.ATTR_IMAGE_CODE);
+        Map<String, Object> dates = (Map<String, Object>) attrMap.get("dates");
+        attrMap.remove("dates");
+        attrMap.put(PackDateDao.PD_DATE_BEGIN, Utils.iso8601Format.parse((String) dates.get("startDate")));
+        attrMap.put(PackDateDao.PD_DATE_END, Utils.iso8601Format.parse((String) dates.get("endDate")));
```
Comment (cesarsantiagonunez) [inline cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java:115]: estas fechas ahora creo que irian contra a taboa pack_date

**Review by cesarsantiagonunez — CHANGES_REQUESTED** [review]

File: cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java (hunk 4 above)
Comment (dari0rm) [reply cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/pack/PackService.java:117]: Ya hablamos con el grupo 2 que fue el encargado de ese método y nos dijeron que como habían cambiado el mismo método, que pusheasemos nuestro cambio igualmente y ellos al hacer el merge de su rama con los cambios del método ya dejarían solo lo suyo.

File: cd2024bfs2g1-frontend/src/main/ngx/src/app/main/pack/pack-new/pack-new.component.html [hunk 5]
```diff
@@ -1,9 +1,9 @@
 <o-form attr="packNew" label-header-align="start" message-service-type="customMessageServiceTypePacks"
         label-header="{{ 'PACKS_NEW' | oTranslate }}" keys="pck_id" fxLayout="column" #oFormPackNew service="packs" entity="pack"
-        keys-sql-types="integer" (onInsert)="insertPacks()" confirm-exit="false" class="custom-form">
-  <o-row layout-align="start stretch">
-    <o-column id="pack" layout-align="start stretch" fxFlex="50" layout-padding>
-      <o-text-input id="text" attr="pck_name" max-length="50" label="{{ 'PACK_NAME' | oTranslate }}"
+     keys-sql-types="integer" (onInsert)="insertPacks($event)" confirm-exit="false" class="custom-form">
```
Comment (cesarsantiagonunez) [inline cd2024bfs2g1-frontend/src/main/ngx/src/app/main/pack/pack-new/pack-new.component.html:3]:
```suggestion
     keys-sql-types="integer" (onInsert)="insertPacks()" confirm-exit="false" class="custom-form">
```

**Review by cesarsantiagonunez — CHANGES_REQUESTED** [review]

**Review by cesarsantiagonunez — APPROVED** [review]

**Pull request merged**